internal/index/indexoperations/index.go

```diff
@@ -52,3 +53,14 @@ func ListIndexes(path string) ([]Index, error) {
 	}
 	return indexes, nil
 }
+
+// AddIndex initializes a new index to install plugins from.
+func AddIndex(path, name, url string) error {
+	dir := filepath.Join(path, name)
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		return gitutil.EnsureCloned(url, dir)
```

Comment:
If cloning fails, you should definitely clean up the directory. Otherwise then index registry is left in an inconsistent state and the user has to clean up manually.

Reply:
Actually, wouldn't the directory not exist if cloning fails?

Reply:
You're right. Git is already taking care of this. My bad.